### ISSUE TITLE
Generalized combineFetchAssoc()

### ISSUE THREAD

**Issue by mkroon1** [LOVDnl/LOVD3#37]
As discussed in PR #32, there's potential for a generalized fetchAllCombine() in the LOVD_PDO class. That would return an array with the selected column as key and an array of all other columns as value.

E.g. for query `SELECT name, id, orcid_id FROM users`, `fetchAllCombine(1)` would return something like:

    array(
        id1 => array(name1, orcid_id1),
        id2 => array(name2, orcid_id2)
    );

**Comment by ifokkema** [comment]
I think you meant to switch name and id in your example, right? So select id first, then the other columns?
Two other things to consider:
- We should make sure we're not breaking existing implementations of it. Also because its current implementation mirrors php's array_combine() function, hence the name. So besides your example, it should keep it's current functionality when only two columns are requested:
```$zUsers = $_DB->query('SELECT id, name FROM lovd_users')->fetchAllCombine();```
should return:
```
array(
    '00001' => 'Name 1',
    '00002' => 'Name 2',
)
```
- We'd want to use keys in the resulting array I think? So your example would return:
```
array(
    id1 => array('name' => name1, 'orcid_id' => orcid_id1),
    id2 => array('name' => name2, 'orcid_id' => orcid_id2)
);
```
or not?

**Comment by mkroon1** [comment]
> I think you meant to switch name and id in your example, right? So select id first, then the other columns?

Actually no, I switched in on purpose, as you can give an index to `fetchAllCombine($col)` for the column you want as keys. Normally you would indeed select id as the first field in the query and call `fetchAllCombine(0)`.

>  Are you intending to create a new function, or to edit fetchAllCombine?

I hadn't decided on that yet, as you say it's vital not to break current uses.

I think you're right that it should return a key and associative array per row.

**Comment by ifokkema** [comment]
> > I think you meant to switch name and id in your example, right? So select id first, then the other columns?

> Actually no, I switched in on purpose, as you can give an index to fetchAllCombine($col) for the column you want as keys. Normally you would indeed select id as the first field in the query and call fetchAllCombine(0).

But... when we will ever use that? We write our own queries. The code will be more complex if we'll allow an arbitrary column as the key, while we can just put the arguments in the right order in the query to begin with. Much simpler :wink:

> > Are you intending to create a new function, or to edit fetchAllCombine?

> I hadn't decided on that yet, as you say it's vital not to break current uses.

It would be nice to have the same function I guess, but then we'll need to make sure that when only two columns are selected, that we return an array as it is now.

**Comment by mkroon1** [comment]
> But... when we will ever use that? We write our own queries.

No, I agree with you completely. It's just that the current fetchAllCombine function allows you to select a column by index, so I didn't want to deviate from that. But if it's going to be a separate function (which I expect) then it would be better to force the caller to select his fields in the right order.

**Comment by ifokkema** [comment]
Ah, right, I forgot I had enabled that option... it was meant for system queries, like:
```$aEngines = $_DB->query('SHOW ENGINES')->fetchAllCombine(0, 1);```
(which just happens to have (0, 1) in use, but it could be (0, 4) or whatever)

Yes, we'll need a separate function indeed :)

**Comment by ifokkema** [comment]
I just noticed something incredibly interesting... this feature already exists in PDO:
```
$q->fetchAll(PDO::FETCH_GROUP | PDO::FETCH_UNIQUE | PDO::FETCH_ASSOC);
```
Does exactly what we want. Since it's so simple, I'm implementing it now, because I need it in the geneloader project :wink:
In the line of `fetchAllAssoc`, `fetchAllColumn`, `fetchAllRow`, `fetchAssoc` and `fetchRow`, I propose to call it `fetchAllGroupAssoc`. What do you think?

**Comment by mkroon1** [comment]
:+1: 

That's perfect, it even converts the id values to integers for us.

**Comment by mkroon1** [comment]
Fixed in #41
